python/ray/train/_internal/logging.py

```diff
@@ -0,0 +1,143 @@
+import logging
+import logging.config
+import os
+from typing import Optional
+
+import yaml
+
+import ray
+from ray.train.constants import (
+    DEFAULT_RAY_TRAIN_LOG_CONFIG_PATH,
+    RAY_TRAIN_JSON_LOG_ENCODING_FORMAT,
+    RAY_TRAIN_LOG_CONFIG_PATH_ENV,
+    RAY_TRAIN_LOG_ENCODING_ENV,
+)
+
+# Env. variable to specify the encoding of the file logs when using the default config.
+RAY_TRAIN_LOG_ENCODING = os.environ.get(RAY_TRAIN_LOG_ENCODING_ENV, "").upper()
+# Env. variable to specify the logging config path use defaults if not set
+RAY_TRAIN_LOG_CONFIG_PATH = os.environ.get(RAY_TRAIN_LOG_CONFIG_PATH_ENV)
```

Comment:
We should get these dynamically rather than at the global level. If the user set an an environment variable in their code, these values wouldn't be updated.

Reply:
+1 Ended up making this change to the ray data code. Defined the name of the variable [here](https://github.com/ray-project/ray/blob/d5fa9a04ed841ea845887f43e06a0d2a81216c2d/python/ray/data/_internal/logging.py#L19) and dynamically loaded [here](https://github.com/ray-project/ray/blob/d5fa9a04ed841ea845887f43e06a0d2a81216c2d/python/ray/data/_internal/logging.py#L116).